peps/pep-0761.rst

```diff
@@ -207,9 +207,57 @@ During the `pre-PEP discussion <https://discuss.python.org/t/pre-pep-discussion-
 there was a question of whether offline verification was supported by
 Sigstore. Using a Sigstore bundle (:file:`.sigstore`) file, `Sigstore clients
 support verifying the artifact completely offline <https://discuss.python.org/t/pre-pep-discussion-stop-providing-gpg-signatures-for-cpython-artifacts/65058/9>`_.
-When in offline mode, Sigstore can't verify whether a signature has been
-revoked. This is a similar restriction to PGP key revocations not being
-detectable during offline verification.
+
+Using offline verification with Sigstore requires disabling root of trust
+updates and "pinning" a root of trust in a file to use during verification.
+
+Pinning a root of trust means signatures made after a *new* root of trust
+is established would no longer be able to verify using a "pinned" previous
+root of trust. New roots of trust are expected to be rare events, such as
+when the root of trust is compromised, and in this case verifiers would
+want signatures to fail to verify.
```

Comment:
I can't follow the reasoning here and wonder if I'm missing something.

In the event of a compromised root of trust, verifiers would want signatures made by the compromised root to fail to verify. But that's not what's happening here: If an offline verifier pinned the compromised root, new signatures of that root (presumably made by a bad actor) will still verify.

What will fail is verification of signatures made by the new root. For these, the offline verifier would want to be able to verify them. Which is not possible, for obvious reasons.

Note that I don't disagree with the overall statement. It's just the sub-clause "in this case verifiers would
want signatures to fail to verify" that I doubt.

(This is just a drive-by comment by a curious reader. I'm not directly affected by this PEP at all.)